capability-providers/keyvalue/rust/src/rpc.rs

```diff
@@ -0,0 +1,181 @@
+use crate::deserialize;
+use crate::{Invocation, InvocationResponse};
+use redis::Connection;
+use redis::{self, Commands};
+use std::error::Error;
+
+const OP_ADD: &str = "Add";
+const OP_GET: &str = "Get";
+const OP_SET: &str = "Set";
+const OP_DEL: &str = "Del";
+const OP_CLEAR: &str = "Clear";
+const OP_RANGE: &str = "Range";
+const OP_PUSH: &str = "Push";
+const OP_LIST_DEL: &str = "ListItemDelete";
+const OP_SET_ADD: &str = "SetAdd";
+const OP_SET_REMOVE: &str = "SetRemove";
+const OP_SET_UNION: &str = "SetUnion";
+const OP_SET_INTERSECT: &str = "SetIntersection";
+const OP_SET_QUERY: &str = "SetQuery";
+const OP_KEY_EXISTS: &str = "KeyExists";
+
+use crate::generated::*;
+
+type OpResult = Result<InvocationResponse, Box<dyn Error + Send + Sync>>;
+
+pub(crate) fn handle_rpc(inv: Invocation) -> InvocationResponse {
+    // We can assume all origins of calls coming to this provider
+    // are from actors
+    match dispatch_operation(&inv.id, &inv.operation, &inv.origin.public_key, &inv.msg) {
+        Ok(v) => v,
+        Err(e) => InvocationResponse::failure(&inv, &format!("{}", e)),
+    }
+}
+
+fn dispatch_operation(inv_id: &str, op: &str, actor: &str, bytes: &[u8]) -> OpResult {
+    let mut conn = actor_con(actor)?;
+    let inv = match op {
+        OP_ADD => add(&mut conn, deserialize(bytes)?),
+        OP_DEL => del(&mut conn, deserialize(bytes)?),
```

Comment:
Just a thought for codegen later, this feels like something we can absolutely generate scaffolding for. With the list of operations defined, go ahead and generate the long match and let people implement the individual functions.